pkg/kube-plex/controller_test.go

```diff
@@ -0,0 +1,18 @@
+package kubeplex
+
+import (
+	"testing"
+	fakeclientset "github.com/munnerz/kube-plex/pkg/client/clientset/versioned/fake"
+	fakekubernetes "k8s.io/client-go/kubernetes/fake"
+)
+
+func TestController(t *testing.T) {
+	ptj := GeneratePlexTranscodeJob([]string{"/hello", "world"})
+
+	kc := KubeClient{
+		Clientset: fakeclientset.NewSimpleClientset(ptj),
```

Comment:
I haven't been able to figure out how to use these fake objects that the client creates for mocks:

```
# github.com/munnerz/kube-plex/pkg/kube-plex
pkg/kube-plex/controller_test.go:13:12: cannot use "github.com/munnerz/kube-plex/pkg/client/clientset/versioned/fake".NewSimpleClientset(ptj) (type *"github.com/munnerz/kube-plex/pkg/client/clientset/versioned/fake".Clientset) as type *kubernetes.Clientset in field value
pkg/kube-plex/controller_test.go:13:46: cannot use ptj (type "github.com/munnerz/kube-plex/pkg/apis/ptj/v1".PlexTranscodeJob) as type "github.com/munnerz/kube-plex/vendor/k8s.io/apimachinery/pkg/runtime".Object in argument to "github.com/munnerz/kube-plex/pkg/client/clientset/versioned/fake".NewSimpleClientset:
	"github.com/munnerz/kube-plex/pkg/apis/ptj/v1".PlexTranscodeJob does not implement "github.com/munnerz/kube-plex/vendor/k8s.io/apimachinery/pkg/runtime".Object (DeepCopyObject method has pointer receiver)
pkg/kube-plex/controller_test.go:14:17: cannot use "github.com/munnerz/kube-plex/vendor/k8s.io/client-go/kubernetes/fake".NewSimpleClientset() (type *"github.com/munnerz/kube-plex/vendor/k8s.io/client-go/kubernetes/fake".Clientset) as type versioned.Interface in field value:
	*"github.com/munnerz/kube-plex/vendor/k8s.io/client-go/kubernetes/fake".Clientset does not implement versioned.Interface (missing Kubeplex method)
FAIL	github.com/munnerz/kube-plex/pkg/kube-plex [build failed]
```

Once I figure that out I can write unit tests for all this.

Reply:
Figured this out, I needed to use the kubernetes.Interface instead of kubernetes.Clientset and then I also had the arguments reversed when instantiating it..